src/lib/state-incentives-calculation.ts

```diff
@@ -129,6 +136,21 @@ export function calculateStateIncentivesAndSavings(
       }
     }
 
+    // check if the incentive is a tax credit and the users's tax liability is less than the credit amount
+    if (
+      item.payment_methods[0] === 'tax_credit' &&
+      typeof stateTaxOwed.tax_owed === 'number' &&
+      stateTaxOwed.tax_owed < item.amount.number
+    ) {
+      // mark the incentive ineligible if the user owes no taxes and the incentive isn't redeemable in another way
+      if (stateTaxOwed.tax_owed === 0 && item.payment_methods.length === 1) {
+        eligible = false;
+      }
+
+      // change the tax credit amount to match the user's max tax liability
+      item.amount.number = stateTaxOwed.tax_owed;
```

Comment:
I'm a little hesitant about this. AFAIK we're currently always presenting incentives with non-customized amounts (though correct me if I'm wrong!) so this would be breaking new ground.

It could result in showing incentives with weird non-round numbers, and it also might not be clear to the user why we're showing an amount that's different from the amount in the incentive's official definition.

It's also sort of an incomplete version of what it's trying to do: suppose you're eligible for two tax credits of $1000 each, and your tax liability is $800. This will set both tax credits' amounts to $800, and you'll see that you're eligible for two $800 credits. But you're still only going to be able to claim $800 worth of credits total, not $1600.

I do think it's valuable to _somehow_ communicate when a user can't actually claim the total value of all the tax credits we show them, but I think it needs to be a deeper rethink.

Reply:
Yeah, that's fair. I'll revert it!